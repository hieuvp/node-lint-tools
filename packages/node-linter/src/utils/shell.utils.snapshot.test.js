const { decorate } = require('./shell.utils');

describe('decorate', () => {
  [
    'ls',
    'ls -lia',
    'which atom',
    'brew cask outdated',
    'docker info | grep -E "Containers|Images"',
    '/usr/local/bin/npm -g outdated --parseable --depth=0',
    'ls "/Users/hieuvp/Library/Preferences" | grep "^IntelliJIdea"',
    'tree /Volumes/Data/Workspace -L 2 > "/Volumes/Data/Dropbox/Backup/Workspace.txt"',
    'cd /Volumes/Data/Workspace/Farmers/logtan-api && git branch placeholder && git checkout placeholder'
  ].forEach(command => {
    it(`should match snapshot with command: "${command}"`, () => {
      expect(decorate(command)).toMatchSnapshot();
    });
  });
});
