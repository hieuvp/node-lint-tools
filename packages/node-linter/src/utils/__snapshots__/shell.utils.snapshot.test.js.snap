// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decorate should match snapshot with command "/usr/local/bin/npm -g outdated --parseable --depth=0" 1`] = `
"
=========================================================
 $ /usr/local/bin/npm -g outdated --parseable --depth=0
=========================================================
"
`;

exports[`decorate should match snapshot with command "brew cask outdated" 1`] = `
"
=======================
 $ brew cask outdated
=======================
"
`;

exports[`decorate should match snapshot with command "cd /Volumes/Data/Workspace/Farmers/logtan-api && git branch placeholder && git checkout placeholder" 1`] = `
"
========================================================================================================
 $ cd /Volumes/Data/Workspace/Farmers/logtan-api && git branch placeholder && git checkout placeholder
========================================================================================================
"
`;

exports[`decorate should match snapshot with command "docker info | grep -E "Containers|Images"" 1`] = `
"
==============================================
 $ docker info | grep -E \\"Containers|Images\\"
==============================================
"
`;

exports[`decorate should match snapshot with command "ls "/Users/hieuvp/Library/Preferences" | grep "^IntelliJIdea"" 1`] = `
"
==================================================================
 $ ls \\"/Users/hieuvp/Library/Preferences\\" | grep \\"^IntelliJIdea\\"
==================================================================
"
`;

exports[`decorate should match snapshot with command "ls -lia" 1`] = `
"
============
 $ ls -lia
============
"
`;

exports[`decorate should match snapshot with command "ls" 1`] = `
"
=======
 $ ls
=======
"
`;

exports[`decorate should match snapshot with command "tree /Volumes/Data/Workspace -L 2 > "/Volumes/Data/Dropbox/Backup/Workspace.txt"" 1`] = `
"
=====================================================================================
 $ tree /Volumes/Data/Workspace -L 2 > \\"/Volumes/Data/Dropbox/Backup/Workspace.txt\\"
=====================================================================================
"
`;

exports[`decorate should match snapshot with command "which atom" 1`] = `
"
===============
 $ which atom
===============
"
`;
